app/(admin)/admin.tsx

```diff
@@ -1,56 +1,277 @@
+    const [team2ID, setTeam2ID] = useState('');
+    const [date, setDate] = useState(new Date());
+    // create team states
+    const [createTeamModalVisible, setCreateTeamModalVisible] = useState(false);
+    const [teamName, setTeamName] = useState('');
+    const [color1, setColor1] = useState('');
+    const [color2, setColor2] = useState('');
+    const [logo, setLogo] = useState('');
+    //  edit team states
+    const [editTeamModalVisible, setEditTeamModalVisible] = useState(false);
+    const [oldTeamID, setOldTeamID] = useState('');
+    const [newTeamName, setNewTeamName] = useState('');
+    const [newColor1, setNewColor1] = useState('');
+    const [newColor2, setNewColor2] = useState('');
+    const [newLogo, setNewLogo] = useState('');
+    
+    const [isDatePickerVisible, setDatePickerVisibility] = useState(false);
+    const [isTimePickerVisible, setTimePickerVisibility] = useState(false);
 
-    }
+    const showDatePicker = () => {
+        setDatePickerVisibility(true);
+    };
 
-    const editGame = async () => {
-        
-    }
+    const showTimePicker = () => {
+        setTimePickerVisibility(true);
+    };
 
-    const deleteGame = async () => {
-        
-    }
+    const onChangeDate = (event: DateTimePickerEvent, selectedDate?: Date) => {
+        setDatePickerVisibility(false); // Hide the date picker after selection
+        const currentDate = selectedDate || date;
+        setDate(currentDate);
+    };
 
-    const createTeam = async () => {
-        
+    const onChangeTime = (event: DateTimePickerEvent, selectedTime?: Date) => {
+        setTimePickerVisibility(false); // Hide the time picker after selection
+        const currentTime = selectedTime || date;
+        setDate(currentTime);
+    };
+    const handleAddGame = async (db: Firestore, team1ID: String, team2ID: String, startTime: Timestamp) => {
+        if (team1ID && team2ID && startTime) {
+    
+            const docRef = await addDoc(collection(db, "games"), {
+                gameID: '',
+                gameState: "inactive", 
+                currentQuestion: 0,
+                questions: [],
+                startTime: startTime,
+                team1ID: team1ID,
+                team2ID: team2ID, 
+                team1score: 0, 
+                team2score: 0,
+            }); 
+        } else {
+            alert("Make sure all fields are filled!")
+        }
     }
 
-    const editTeam = async () => {
-        
+    const handleAddTeam = async (db: Firestore, teamName: String, color1: String, color2: String, logo: String) => {
+        if (teamName && color1 && color2 && logo) {
+            const docRef = await addDoc(collection(db, "teams"), {
+                teamID: '',
+                teamName: teamName,
+                color1: color1, 
+                color2: color2,
+                logo: logo
+            });
+
+        } else {
+            alert("Make sure all fields are filled!")
+        }
```

Comment:
Is there a way that we're updating the teamID field in the Firestore? when i made a team it still had a blank teamID